flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -212,7 +212,10 @@ private static HiveConf createHiveConf(
             @Nullable String hiveConfDir, @Nullable String hadoopConfDir) {
         // create HiveConf from hadoop configuration with hadoop conf directory configured.
         Configuration hadoopConf = null;
-        if (isNullOrWhitespaceOnly(hadoopConfDir)) {
+        if (!isNullOrWhitespaceOnly(hadoopConfDir)) {
+            hadoopConf = getHadoopConfiguration(hadoopConfDir);
+        }
+        if (hadoopConf == null) {
```

Comment:
@hehuiyuan , if the `hadoopConfDir ` has been configured and the `hadoopConf` is null, it should create new `Configuration`. Otherwise, it causes that the `hadoopConfDir` configured looks like correct for the users.

Reply:
Hi, make some changes.